### ISSUE TITLE
[ITEM][Alchemy] Oil of Immolation

### ISSUE THREAD

**Issue by balkai** [dalaranwow/dalaran-wow#4454]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
Oil of Immolation currently shares a hidden CD with all other pots, hidden meaning its visually not showing the CD but you can't drink it and error shows up "You can't use that yet"

**Expected behaviour**: Tell us what should happen instead
Oil of Immolation should only share CD with elixirs/flasks, which is a 3 second cd and is not affected by combat.

**Steps to reproduce the problem**:

1. Enter combat, drink Oil of Immolation
2. Wait for 3 second CD
3. Try to drink another, you get error message.

**Include proofs for this behaviour**

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**
Item ID: 8956
http://wotlk.openwow.com/item=8956
list of items it shares CD with
http://www.wowhead.com/item=8956/oil-of-immolation#shared-cooldown

You can research this on several pages like http://wotlk.openwow.com or http://wowhead.com/

**Include Screenshots from the issue if necessary**

**Comment by Rushor** [comment]
In general you should not be able to cast any potion for a second time if you are in combat. 
That#s how it's implemented currently:

```c++
// check if we are using a potion in combat for the 2nd+ time. Cooldown is added only after caster gets out of combat
            if (m_caster->ToPlayer()->GetLastPotionId() && m_CastItem && (m_CastItem->IsPotion() || m_spellInfo->IsCooldownStartedOnEvent()))
                return SPELL_FAILED_NOT_READY;
```

This prevents several abuses. 

Please provide a source (video/wowwiki etc.) that Oil of Immolation can be used several times in combat and i will reopen and fix the issue. Thank you.

(Setting the issue "on hold" to wait for proof sources)

**Comment by balkai** [comment]
It's not on the list of items with shared CD of potions, it's not considered a potion either
http://wowwiki.wikia.com/wiki/Potion
Oil of Immolation has always been an exception, was more relevant in vanilla, but used in Wotlk as a min/max for dps.

http://www.wowhead.com/item=8956/oil-of-immolation#comments
By Bloodmuffin (176 – 1) on 2009/10/27 (Patch 3.2.2)		
"For Engineers, a simple macro for min/maxing AoE between abilties:

#showtooltip Saronite Bomb
/use Oil of Immolation
/use Saronite Bomb

Yeah, it won't do much more damage, but it doesn't burn the GCD so it's kinda nice to use, if the cost is cheap, and you can afford the bag space"

I can't find a video that is not from vanilla

**Comment by TheRealestMcCoy** [comment]
Here is some more research for this.
NotesEdit
Oil of Immolation has a number of distinct uses: in PvE, it helps the Tank to gain extra threat on multiple targets (since the damage accumulates and is actually quite significant compared to things like Retribution Aura, Naglering and so on). Also, a raid group can cause a significant amount of extra AoE in a confined area, such as when fighting High Priestess Arlokk and her panthers. In PvP, it is useful for spotting hidden rogues, for classes that don't have AoE or stealth detection. Never forget to drink your Oil of Immolation when approaching a seemingly unguarded flag!
**Does not share cooldown with other potions and can be chugged in combat as often as every 3 seconds.**

[Source](http://wowwiki.wikia.com/wiki/Oil_of_Immolation)

**Comment by plznoz** [comment]
Might be worth checking if it scales with spellpower currently (it shouldn't).
> Patch 1.6.1 (02-Aug-2005): The Oil of Immolation effect is no longer affected by bonuses to spell damage from items.

http://wowwiki.wikia.com/wiki/Oil_of_Immolation

**Comment by Rushor** [comment]
adding a flag extra that ignores the potion check should fix this to my mind

**Comment by balkai** [comment]
It doesnt scale with spellpower, its a static 50 dmg, can crit tho, which is how its supposed to work, crits static dmg too

**Comment by Rushor** [comment]
on ptr it scales with spellpower

**Comment by plznoz** [comment]
My friend told me it did scaled with spellpower on live at some point. Haven't checked again.

**Comment by Rushor** [comment]
Game/Spells: Implement ITEM_FLAGS_CU_IGNORE_POTION_CHECK to ignore the potioncheck of 'Oil of Immolation' + remove spellbonus for 'Fire Shield'

* closes: https://github.com/dalaranwow/dalaran-wow/issues/4454
* commit: d1f0169a3b75539488b9dd2567c823113e82b8c7

The issue was fixed by a developer and will now be tested on the Public Test Realm (PTR). The changes/fixes will be applied with the NEXT changelog on the liverealm.
This means that the issue exists in game at the moment i wrote this message.
Only once the issue was marked with "Fixed - On Live Server", you can find the fix on you server.
Please watch http://www.dalaran-wow.com/news/ to see when the changelog and the fix was applied.

**Comment by TheRealestMcCoy** [comment]
It should only scale with fire talents. Not Spell power.

**Comment by TheRealestMcCoy** [comment]
Here is a comment from wowhead.

By M3ss3ng3r (1,389 – 17) on 2009/07/13 (Patch 3.1.3)		
I decided to test Oil of Immolation to see if it benefitted from mages' Fire talents, after finding that the Midsummer AoE damage buff could trigger the Impact talent.

Tested on dummies in UC using my mage. He simply used an Oil and stood next to the dummies without attacking or doing anything else. Oil of Immolation can indeed trigger Impact. I also noticed an unusually high incidence of crits (75 damage rather than 50), so it's possible any crit bonuses you have are also applied to Oil of Immolation.

Sadly, because damage remained at a constant 50 (or 75) as described in its tooltip, Oil of Immolation does not benefit from spellpower bonuses.

Guessing that fire-based talents from other classes will work on Oil of Immolation, this info can be useful for shamans and warlocks.
